src/library/scala/collection/Map.scala

```diff
@@ -207,6 +211,16 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
   }
 
   /** An [[Iterable]] collection of the keys contained by this map.
+   *
+   *  The default implementation returns `keySet`.
```

Comment:
OK, but then my next question as a user is: when does it not return `keySet`? When should I use this over `keySet`?

Reply:
They must read the user manual for their implementation.